Lib/test/test_httpservers.py

```diff
@@ -315,6 +332,76 @@ def test_head_via_send_error(self):
             self.assertEqual(b'', data)
 
 
+@unittest.skipIf(ssl is None, 'No ssl module')
+class BaseHTTPSServerTestCase(BaseTestCase):
+    def _data_file(*name):
+        return os.path.join(os.path.dirname(__file__), "certdata", *name)
+
+    CERTFILE = _data_file("keycert.pem")
+    ONLYCERT = _data_file("ssl_cert.pem")
+    ONLYKEY = _data_file("ssl_key.pem")
+    CERTFILE_PROTECTED = _data_file("keycert.passwd.pem")
+    ONLYKEY_PROTECTED = _data_file("ssl_key.passwd.pem")
+    KEY_PASSWORD = "somepass"
+    EMPTYCERT = _data_file("nullcert.pem")
+    BADCERT = _data_file("badcert.pem")
+
+    tls = (ONLYCERT, ONLYKEY, None) # values by default
+
+    class request_handler(NoLogRequestHandler, SimpleHTTPRequestHandler):
+        pass
+
+    def test_get(self):
+        response = self.request('/')
+        self.assertEqual(response.status, HTTPStatus.OK)
+
+    def request(self, uri, method='GET', body=None, headers={}):
+        self.connection = http.client.HTTPSConnection(
+            self.HOST,
+            self.PORT,
+            context=ssl._create_unverified_context()
+        )
+        self.connection.request(method, uri, body, headers)
+        return self.connection.getresponse()
+
+    def test_valid_certdata(self):
+        valid_certdata_examples = (
+            (self.CERTFILE, None, None),
+            (self.CERTFILE, self.CERTFILE, None),
+            (self.CERTFILE_PROTECTED, None, self.KEY_PASSWORD),
+            (self.ONLYCERT, self.ONLYKEY_PROTECTED, self.KEY_PASSWORD),
+        )
+        for data in valid_certdata_examples:
+            server = HTTPSServer(
+                ('localhost', 0),
+                BaseHTTPRequestHandler,
+                certfile=data[0],
+                keyfile=data[1],
+                password=data[2],
+            )
+            self.assertIsInstance(server, HTTPSServer)
```

Comment:
```suggestion
        valid_certdata= [
            (self.CERTFILE, None, None),
            (self.CERTFILE, self.CERTFILE, None),
            (self.CERTFILE_PROTECTED, None, self.KEY_PASSWORD),
            (self.ONLYCERT, self.ONLYKEY_PROTECTED, self.KEY_PASSWORD),
        ]
        for certfile, keyfile, password in valid_certdata:
            server = HTTPSServer(
                ('localhost', 0),
                BaseHTTPRequestHandler,
                certfile=certfile,
                keyfile=keyfile,
                password=password,
            )
            self.assertIsInstance(server, HTTPSServer)
```

Ideally, I think we can have a function helper for 

```
HTTPSServer(
    ('localhost', 0),
    BaseHTTPRequestHandler,
    certfile=certfile,
    keyfile=keyfile,
    password=password,
)
```

with defaults here and there (like the handler=BaseHTTPRequestHandler and `address=('localhost', 0)` but maybe in a follow-up cleanup PR if it can be reused elsewhere.